### ISSUE TITLE
Android Back navigation - possible bug

### ISSUE THREAD

**Issue by bFlood** [eclipsesource/tabris-js#733]
in Android, if you navigate to a page and then very quickly hit the back button (and less frequently, the navbar back button), it will leave the previous page blank. It's navbar is there but the content is not visible. maybe some animation is not triggered??

**Comment by cookieguru** [comment]
Debug or production build?  I've noticed the page stacks work a little differently but haven't been able to pinpoint the cause.

**Comment by bFlood** [comment]
we can repo in release and debug

**Comment by cookieguru** [comment]
Do you have a snippet?  Are either or both of them top level pages?

**Comment by bFlood** [comment]
the initial page is top level, the navigated to page is not (for our example of whne we've seen it, I'll try to find more examples). it doesn't matter the page or the amount of controls on it, its pretty easy to reproduce now

edit - I just tried on the tabris.js bookstore example and I can see the same thing happening. click on a book in the collectionview and then quickly hit the back button, initial page with the collectionview is blank

**Comment by cookieguru** [comment]
I tried hard to repro this with the Bookstore example running in the Tabris.js 1.5 developer app on a pair of Samsung 4.4 devices but couldn't.  I hit the back button just as soon as I could get my finger to the back button after tapping an item in the collection.  Sometimes this caused me to jump back in to the examples list presumably because the new page hadn't finished being initialized so the top level listing page was the only one on the stack.  I also tried waiting a few hundred milliseconds after tapping the item, but no dice.

If I tapped on two or three items in the collection in quick succession I would get two/three pages (respectively) to open, each with the correct data.  These three pages would get pushed on to the stack and would pop off as expected.  But never a blank page.

Can you run this snippet and see what gets logged when you have the blank page?  And if it's undefined, it would be interesting to see what's in `tabris.ui._pages`.  I think what might be happening is that you pop the last page off before the new page can get added to the stack.

```js
tabris.create('Page', {
  topLevel: true,
  title: 'Top level page'
}).append(
  tabris.create('Button', {
    text: 'Open child page'
  }).on('select', function() {
    tabris.create('Page', {
      title: 'Child page'
    }).open();
  })
).open();

tabris.ui.on('change:activePage', function(ui, page) {
  console.log('Now active: ', page.get('title'));
});
```

**Comment by bFlood** [comment]
here's a quick video of your script and the bookstore demo app
https://drive.google.com/file/d/0B70iZVVL69Kaa29VcGczb2JkMVE/view?usp=sharing

Android 5.0.1 but also seen on Android 5+ devices as well. I have not tried on 4.4

**Comment by bFlood** [comment]
fwiw, I think is has to do with the animation when you go back to the previous screen, its either not started or canceled. if you look at the video of the bookstore, you can see the animation continue after the page goes blank just before the click on the gear action button on the navbar

**Comment by cookieguru** [comment]
Good catch, it does indeed have to do with the animation.  Normally I run with all animations disabled (Developer options -> Window animation scale, transition animation scale, animator duration scale) all turned off.  I just set them back to 1x and this repros no problem.

**Comment by bFlood** [comment]
yea, its pretty easy to put your app in an invalid state where users have no idea what to do next. several of our testers have reported this error but I never knew what they were talking about, you really have to go back and forth fairly quickly to see it (then again, that's how real uses interact with apps though)

**Comment by mpost** [comment]
I can confirm the reported issue. We were able to reproduce it in the bookstore demo as shown in the video.

**Comment by mpost** [comment]
The issue has been fixed for the upcoming 1.6 release.
